File: solutions/python/resistor-color-duo/resistor_color_duo.py
"""Helpful program so that you don't have to remember the values of the bands."""


def value(colors):
    """Calculate a resistance value from two colors"""
    return COLOR_CODES.index(colors[0]) * 10 + COLOR_CODES.index(colors[1])


COLOR_CODES = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]

File: solutions/javascript/resistor-color-duo/resistor-color-duo.js
//@ts-check
//
// Helpful program so that you don't have to remember the values of the bands.
//

/** Calculate a resistance value from two colors
 *
 * @param {string[]} colors
 * @return {number}
 */
export const decodedValue = (colors) =>
  COLORS.indexOf(colors[0]) * 10 + COLORS.indexOf(colors[1]);

const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];
